packages/shared/src/guards.ts

```diff
@@ -361,11 +369,17 @@ export function isNdArray<T extends Data>(
 }
 
 export function isTypedArray<T, U extends TypedArray>(arr: U | T[]): arr is U {
-  return isTypedArrayLodash(arr);
+  return !Array.isArray(arr);
```

Comment:
Since this guard only accepts `TypedArray | T[]`, we can assume that anything that is now a plain JS array is a typed array. This is most likely faster than lodash's `isTypedArray`, which runs a regex against the result of `toString()`.